### PR TITLE
Standardized TAP file name

### LINKED ISSUE #236: Standardize TAP file name
Currently, TKG creates TAP file called `TestTargetResult.tap`. Since we are going to collect the TAP files for AQAvit certification, we will collect TAP files for all test builds. It means we can get lots of these TAP files (the more parallel runs that we have, the more files we will have). We need to standardize the TAP file name. This will help us to easily identify the level and group of the tap file and display the results in the Grid view in TRSS. Also, putting this at the TKG level will benefit automated and local runs.

I would suggest creating a TAP file name based on JDK_VERSION, IMPL, TARGET, and PLATFORM info (similar to Jenkins test job name)- Test_openjdk<JDK_VERSION>_<JDK_IMPL>_<TARGET>_<PLATFORM>.tap
i.e., Test_openjdk8_j9_sanity.openjdk_ppc64_aix.tap. 
Also, the user can set this filename. If the value is not provided by the user, TKG should generate the tap file name as above.

### PR DESCRIPTION
**Pull request by amandanguyeen**
Fixes #236 

Format is now: Test_openjdkJDK VERSION_JDK IMPL_TESTTARGET_SPEC.tap
Example: `Test_openjdk17_hotspot_jdk_math_osx_x86-64.tap`
Signed-off-by: Amanda Nguyen <amanda6@ualberta.ca>

### CODE REVIEW

**Comment by amandanguyeen** [comment]
I saw that there was another variable called CUSTOM_TARGET but assumed that the target we wanted was test target.

**Comment by llxia** [comment]
Thanks @amandanguyeen . Yes, we do not need to worry about CUSTOM_TARGET. However, there are a few cases that we need to take care of:
- TARGET can be a list of tests. i.e.,  `testList TESTLIST=jit_jitt,jit_recognizedMethod_1,testSCCMLTests2_1`. When this happens, we should use `testList` in the TAP filename (instead of the value of TARGET). i.e., Test_openjdk17_hotspot_testList_x86-64_mac.tap
- we need to use PLATFORM, not SPEC. That is, x86-64_mac for macos
- user should be able to set TAP file name if needed

**Comment by llxia** [comment]
@amandanguyeen is this ready for review?

**Review by llxia — CHANGES_REQUESTED** [review]

File: scripts/resultsSum.pl [hunk 1]
```diff
@@ -207,6 +210,21 @@ sub resultReporter {
 	print "   SKIPPED: $numOfSkipped";
 	print "\n";
 	if ($numOfTotal > 0) {
+		# set tap file name if not given
+		my @splitTapFile = split('/', $tapFile);
+		my $userTapInput = $splitTapFile[-1];
```
[the remark this answers is not in the document]
Comment (amandanguyeen) [reply scripts/resultsSum.pl:215]: If the tap name is not set then userTapInput will be an empty string which gets handled in the if statement below
Definitely like the idea of having two variables vs splitting, will update!

File: settings.mk [hunk 2]
```diff
@@ -54,6 +54,10 @@ else
 	export JDK_VERSION:=$(JDK_VERSION)
 endif
 
+ifdef TAP_NAME
+	export TAP_NAME:=$(TAP_NAME).tap
```
[the remark this answers is not in the document]
Comment (amandanguyeen) [reply settings.mk:58]: from what I understand I thought a user would have to do something like `export TAP_NAME=blah` in order to set the tap name no?

**Pull request merged**